server/src/main/kotlin/org/elaastic/player/results/learner/LearnerResultsModelFactory.kt

```diff
@@ -1,18 +1,61 @@
 package org.elaastic.player.results.learner
 
-import org.elaastic.player.explanations.ExplanationDataFactory
+import org.elaastic.activity.response.Response
 import org.elaastic.material.instructional.question.ExclusiveChoiceSpecification
 import org.elaastic.material.instructional.question.MultipleChoiceSpecification
+import org.elaastic.material.instructional.question.QuestionType
 import org.elaastic.material.instructional.statement.Statement
-import org.elaastic.activity.response.Response
+import org.elaastic.player.explanations.ExplanationDataFactory
 
 object LearnerResultsModelFactory {
 
-    fun buildOpenResult(responseFirstTry: Response?,
-                        responseSecondTry: Response?,
-                        responseFirstTryHasChatGPTEvaluation: Boolean,
-                        responseSecondTryHasChatGPTEvaluation: Boolean,
-                        ) : LearnerOpenResults =
+    /**
+     * Get the LearnerResultsModel for the given responses and statement
+     *
+     * @param responseFirstAttempt the response of the learner for the first attempt
+     * @param responseSecondAttempt the response of the learner for the second attempt
+     * @param statement the statement of the sequence
+     */
+    fun builtLearnerResultsModel(
+        responseFirstAttempt: Response?,
+        responseSecondAttempt: Response?,
+        responseFirstTryHasChatGPTEvaluation: Boolean,
+        responseSecondTryHasChatGPTEvaluation: Boolean,
```

Comment:
It seems to me that there is a bulk of related data provided as separate parameters here.
I would prefer to introduce a type for this.

For example:
```kotlin
class LearnerSequenceResponses(
    val responseFirstTry: Response?,
    val responseSecondTry: Response?,
    val chatGptEvaluationResponseStore: ChatGptEvaluationResponseStore
) {

    fun getNonNullResponses(): List<Response> = listOfNotNull(responseFirstTry, responseSecondTry)
}
```